content/millennium/bulk-data.md

```diff
@@ -0,0 +1,91 @@
+
+* The size of the population for which data was requested
+* The number of resource types requested
+* The time range of activity being requested
+* Inherent latency due to network issues and effects from concurrent requests to the system
+
+#### Example use cases
+
+* Creating an internal clinical data warehouse for study purposes
+* Consolidation of claims data for comprehensive view
+* Machine learning applications obtaining training data from EHR
+* Integration of population health system with an EHR system
+* Data transfer to payer database for care quality assessment
+* Submission to reportable disease (or other) registries
+
+<!-- TOC -->
+* TOC
+{:toc}
+
+## Implemented Exports
+
+Cerner supports only the Group Export operation.
+
+* [Group Export](/millennium/bulk-data/group-export/)
+
+## Registering Client Applications
+
+In order to be granted access to the Bulk Data Access API, a Millennium Bulk SMART application must be registered through Cerner's [code Console](http://fhir.cerner.com/smart/#code-console). See [Authorization](http://fhir.cerner.com/authorization/) for additional details on registering client applications and authorizing with the server.
+
+In the case of Group Export, client application developers will need to work with healthcare organizations to identify the available Groups to be exported. Organizations can manage Groups within Ignite Management Tooling.
+
+_Notes_
+
+* Authentication is required, open endpoints do not support bulk export workflows.
+* Cerner supports v1.0.1 of the [FHIR Bulk Data Access (FLATFHIR) Implementation Guide](https://hl7.org/fhir/uv/bulkdata/STU1.0.1/)
+* Bulk data runs against an organization's database and utilizes the same services running other applications, so it is important to consider performance. Bulk data exports large amounts of data for large groups of patients, which takes more time to complete the larger the data set or patient population.
+
+## Client Errors
+
+The following are the possible client errors that can be received on API calls:
+
+1. Failing to send a required search parameter or sending a parameter with an invalid value will result in a `400 Bad Request` response.
+
+        HTTP/1.1 400 Bad Request
+
+2. Making a request without valid credentials will result in a `401 Unauthorized` response.
+
+        HTTP/1.1 401 Unauthorized
+
+3. Requesting data from an unknown instance or an instance where the application is not authorized will result in a `403 Forbidden` response.
+
+        HTTP/1.1 403 Forbidden
+
+4. Requesting a resource which does not exist will result in a `404 Not Found` response.
+
+        HTTP/1.1 404 Not Found
+
+5. Requesting a media type other than JSON will result in a `406 Not Acceptable` response.
+
+        HTTP/1.1 406 Not Acceptable
+
+        Content-Length: 0
+
+6. Performing an export kick-off using unsupported search parameter values
+
+        HTTP/1.1 422 Unprocessable Entity
+
```

Comment:
do we also want to add the 429 case?

Reply:
I was thinking about it initially, but since this section is about user errors, I am not sure 429 is applicable. I guess it could be because users can try to resolve it by waiting and sending the request later.

Reply:
generally 4xx errors are user errors. I was thinking it is helpful to document it here so that they know they might get 429 incases when they try to export for the same tenant/client  while another one is in progress

Reply:
429 is part of the errors table here: https://github.com/cerner/fhir.cerner.com/pull/899/commits/eff91ed66335ec7340cb5a2ec8a51663c1bd9cbf#diff-c2a4803bbbf3b17e514d41cc6eb7d578c94a283f937cce0400f5437570744d7fR114

Reply:
Made some changes.
Added it to the common errors:https://github.com/cerner/fhir.cerner.com/pull/899/commits/780d6472fc27b87e37ef7361457f44e8da969b69#diff-b781e82e47439c1cc68cf3a6a51ab1668b5f40d37870022bec1936c5c69d6bc1R94
Present in the errors for kick off request: https://github.com/cerner/fhir.cerner.com/pull/899/commits/780d6472fc27b87e37ef7361457f44e8da969b69#diff-c2a4803bbbf3b17e514d41cc6eb7d578c94a283f937cce0400f5437570744d7fR121